blackbox/docs/best_practice/docker.txt

```diff
@@ -0,0 +1,217 @@
+.. highlight:: sh
+
+.. _cratedb_docker:
+
+==================
+CrateDB and Docker
+==================
+
+CrateDB and Docker_ are a great match thanks to CrateDB's shared-nothing,
+horizontally scalable architecture that lends itself well to containerization.
+
+This document covers the essentials of running CrateDB and Docker using the
+official `CrateDB Docker image`_.
+
+.. NOTE::
+
+   If you're just getting started with CrateDB and Docker, check out our
+   introductory guides for `running Docker locally`_ or `Docker Cloud`_.
+
+Quick Start
+===========
+
+Creating a Cluster
+------------------
+
+To get started with CrateDB and Docker on your dev machine, start by creating a
+`user-defined network`_, like so::
+
+    sh$ docker network create crate
+
+You should then be able to see something like this:
+
+.. code-block:: text
+
+    sh$ docker network ls
+    NETWORK ID          NAME                DRIVER              SCOPE
+    1bf1b7acd66f        bridge              bridge              local
+    51cebbdf7d2b        crate               bridge              local
+    5b8e6fbe9ab6        host                host                local
+    8baa149b6986        none                null                local
+
+Any CrateDB containers put into the ``crate`` network will be able to resolve
+other CrateDB containers by name.
+
+You can then create your first CrateDB container, like so::
+
+    sh$ docker run -d --name=crate01 \
+          --net=crate -p 4201:4200 \
+          crate
+
+Breaking it down, that command:
+
+- Creates and runs a container called ``crate01`` in detached mode
+- Puts the container into the ``crate`` network and maps port ``4201`` on your
+  host machine to port ``4200`` on the container
+- Runs the command ``crate`` daemon inside the container
+
+Check the node is up with ``docker ps`` and you should see something like
+this:
+
+.. code-block:: text
+
+    sh$ docker ps
+    CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                                             NAMES
+    f79116373877        crate               "/docker-entrypoin..."   16 seconds ago      Up 15 seconds       4300/tcp, 5432-5532/tcp, 0.0.0.0:4201->4200/tcp   crate01
+
+You can visit the admin UI in your browser with this URL:
+
+.. code-block:: text
+
+    http://localhost:4201/
```

Comment:
`localhost` does not work with macOS afaik because docker runs inside a VM
IP can be obtained by running `docker-machine ip`